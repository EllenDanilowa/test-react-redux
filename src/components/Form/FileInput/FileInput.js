import React from 'react';
import PropTypes from 'prop-types';
import {FieldWrapper, HiddenInput} from '../Form.styled';
import { Icon, Label } from './FileInput.styled';
import uploadIcon from './assets/upload.svg';

const FileInput = ({name, title, refFunc, accept}) => {


  return (
    <div>
      <FieldWrapper>
        <Label htmlFor={name}>
          <HiddenInput
            id={name}
            name={name}
            accept={accept}
            ref={refFunc}
            type="file"/>
            <Icon src={uploadIcon} alt="Upload a file"/>
          {title}
        </Label>
      </FieldWrapper>
    </div>

  );
};

FileInput.propTypes = {
  accept: PropTypes.string,
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  refFunc: PropTypes.func.isRequired,
  watcher: PropTypes.object
};

export default FileInput;
