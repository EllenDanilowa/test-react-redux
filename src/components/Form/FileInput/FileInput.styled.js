import styled from 'styled-components';
import colors from '../../../styles/colors';
import {Label as FormLabel} from '../Form.styled';

export const Label = styled(FormLabel)`
  cursor: pointer;
  color: ${colors.link}
`;

export const Icon = styled.img`
  display: inline-block;  
  margin-right: 5px;
  width: 18px;

`;
