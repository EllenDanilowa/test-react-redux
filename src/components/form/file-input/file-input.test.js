import React from 'react';
import renderer from 'react-test-renderer';
import FileInput from './file-input';

describe('FileInput', () => {
  let element;
  let title;
  let name;
  let ref;

  const createElement = ({ref, ...props}) => (
    renderer.create(<FileInput refFunc={ref} {...props}/>).toJSON()
  );

  beforeEach(() => {
    title = 'Title';
    name = 'name';
    ref = () => {};
  });

  it('renders component with all defined props', () => {
    const alt = 'alt title';
    element = createElement({name, ref, title, alt});

    expect(element).toMatchSnapshot();
  });

  it('renders component with additional props', () => {
    const accept = 'image/*';
    element = createElement({name, ref, title, accept});

    expect(element).toMatchSnapshot();
  });

  it('renders component with default props', () => {
    element = createElement({name, ref, title});

    expect(element).toMatchSnapshot();
  });
});
