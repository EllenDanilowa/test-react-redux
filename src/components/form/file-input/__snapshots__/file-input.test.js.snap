// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput renders component with additional props 1`] = `
<div
  className="sc-AxirZ eQxQOc"
>
  <label
    className="sc-AxjAm sc-AxhCb eJQEJc"
    htmlFor="name"
  >
    <input
      accept="image/*"
      className="sc-AxiKw hWSlZL"
      id="name"
      name="name"
      type="file"
    />
    <img
      alt="Upload a file"
      className="sc-AxhUy eUgZfS"
      src="upload.svg"
    />
    Title
  </label>
</div>
`;

exports[`FileInput renders component with all defined props 1`] = `
<div
  className="sc-AxirZ eQxQOc"
>
  <label
    className="sc-AxjAm sc-AxhCb eJQEJc"
    htmlFor="name"
  >
    <input
      className="sc-AxiKw hWSlZL"
      id="name"
      name="name"
      type="file"
    />
    <img
      alt="alt title"
      className="sc-AxhUy eUgZfS"
      src="upload.svg"
    />
    Title
  </label>
</div>
`;

exports[`FileInput renders component with default props 1`] = `
<div
  className="sc-AxirZ eQxQOc"
>
  <label
    className="sc-AxjAm sc-AxhCb eJQEJc"
    htmlFor="name"
  >
    <input
      className="sc-AxiKw hWSlZL"
      id="name"
      name="name"
      type="file"
    />
    <img
      alt="Upload a file"
      className="sc-AxhUy eUgZfS"
      src="upload.svg"
    />
    Title
  </label>
</div>
`;
